addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/serial/ValloxSerialInterface.java

```diff
@@ -0,0 +1,453 @@
+ * streams to the RS485 vallox interface.
+ *
+ * Main telegram specification taken from C-Code from
+ * https://github.com/windkh/valloxserial
+ *
+ * @author Hauke Fuhrmann - Initial contribution
+ *
+ */
+public class ValloxSerialInterface {
+
+    public static int MINUTE = 60000;
+
+    private Logger logger = LoggerFactory.getLogger(ValloxSerialHandler.class);
+
+    byte senderID = ValloxProtocol.ADDRESS_PANEL8; // we send commands in the name of panel8 (29)
+    byte receiverID = ValloxProtocol.ADDRESS_PANEL1; // we always listen for the telegrams between the master and the
+                                                     // panel1!
+
+    private OutputStream outputStream;
+    private InputStream inputStream;
+    private Socket socket;
+    private ExecutorService listenerExecutor;
+    private ExecutorService heartbeatExecutor;
+    private boolean shutdownListening = false;
+
+    private List<ValueChangeListener> vallueListener;
+    private List<StatusChangeListener> statusListener;
+    private ValloxStore vallox = new ValloxStore();
+
+    private String host;
+    private int port;
+
+    /**
+     * Get a bean with all variable states of the vallox installations. Variables
+     * will be updated after listening has started. Initialially, variables are initialized with
+     * 0 resp. false and might only get updated if the vallox sends updates. Need to
+     * trigger a variable poll explicitly.
+     *
+     * @return the variable store
+     */
+    public ValloxStore getValloxStore() {
+        return vallox;
+    }
+
+    public List<ValueChangeListener> getValueListener() {
+        if (vallueListener == null) {
+            vallueListener = new ArrayList<ValueChangeListener>();
+        }
+        return vallueListener;
+    }
+
+    public List<StatusChangeListener> getStatusListener() {
+        if (statusListener == null) {
+            statusListener = new ArrayList<StatusChangeListener>();
+        }
+        return statusListener;
+    }
+
+    /**
+     * Connect to a TCP providing host that bridges the serial interface of the
+     * vallox installation to a TCP server.
+     *
+     * @param host Hostname or IP Adress of the tcp-server
+     * @param port port of the server
+     * @throws UnknownHostException
+     * @throws IOException
+     */
+    public void connect(String host, int port) throws UnknownHostException, IOException {
+        this.host = host;
+        this.port = port;
+        socket = new Socket(host, port);
+        outputStream = socket.getOutputStream();
+        inputStream = socket.getInputStream();
+        logger.info("Connected to " + host + ":" + port);
+        for (StatusChangeListener l : this.getStatusListener()) {
+            l.statusChanged(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
+        }
+
+    }
+
+    public void reconnect() throws UnknownHostException, IOException {
+        logger.info("Trying to reconnect: " + host + ":" + port);
```

Comment:
Please change the logging level to `debug`. `trace` can be used for even more details, where necessary.